### PR TITLE
[10.x] add digits to `Number::currency()`

### PR DESCRIPTION
**Pull request by emedchill**
This PR adds digits to the `Number::currency()` call to allow for better rounding.

### CODE REVIEW

**Review by driesvints — APPROVED** [review]

File: src/Illuminate/Support/Number.php [hunk 1]
```diff
@@ -103,14 +103,19 @@ public static function percentage(int|float $number, int $precision = 0, ?int $m
      * @param  int|float  $number
      * @param  string  $in
      * @param  string|null  $locale
+     * @param  int|null  $digits
      * @return string|false
      */
-    public static function currency(int|float $number, string $in = 'USD', ?string $locale = null)
+    public static function currency(int|float $number, string $in = 'USD', ?string $locale = null, int $digits = null)
```
Comment (crynobone) [inline src/Illuminate/Support/Number.php:109]: https://en.wikipedia.org/wiki/Fraction

Feel like `$fractions` would be better than `$digits`

Comment (emedchill) [reply src/Illuminate/Support/Number.php:109]: PHP has both fraction and digits in their naming convention (`NumberFormatter::FRACTION_DIGITS`) but I felt that simple `$digits` after the decimal was fine.

I can change it however we want though.

Comment (emedchill) [reply src/Illuminate/Support/Number.php:109]: Maybe follow the other functions and use `$precision`?

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**